tests/telemetry/test_emitter.py

```
@@ -0,0 +1,982 @@
+    @pytest.mark.asyncio
+    async def test_flush_uses_lock(self):
+        """Concurrent flush calls are serialized via lock."""
+        emitter = TelemetryEmitter(endpoint="http://test:8001/events")
+
+        mock_client = AsyncMock()
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_response.raise_for_status = MagicMock()
+
+        # Slow post to simulate long-running flush
+        async def slow_post(*args, **kwargs):
```

Ruff (0.14.13)

Unused function argument: `args`

(ARG001)